pkgs/stdenv/generic/setup.sh

```diff
@@ -995,10 +997,10 @@ configurePhase() {
     fi
 
     if [ -n "$configureScript" ]; then
-        # Old bash empty array hack
-        # shellcheck disable=SC2086
-        local flagsArray=(
-            $configureFlags ${configureFlagsArray+"${configureFlagsArray[@]}"}
+        IFS=" " read -r -a configureFlagsTemp <<< "$configureFlags"
```

Comment:
I'm not super familiar with this, but this looks like it works in the way we want, while avoiding the shellcheck warning.

/cc @ericson2314

Reply:
This might be one of those things that was only an issue pre-Bash4?

Reply:
This is coming from https://github.com/koalaman/shellcheck/wiki/SC2206
it's not so pretty, but it seems to work.

Reply:
Additionally reading up on arrays, I found this, which recommends the same
http://mywiki.wooledge.org/BashGuide/Arrays